metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/sql/DirectSqlSavePartition.java

```diff
@@ -322,6 +322,34 @@ private void _delete(final QualifiedName tableQName, final List<String> partitio
                 rs.getLong("serde_id")));
     }
 
+    /**
+     * Delete all the partitions for the given table <code>tableQName</code>.
+     *
+     * @param tableQName     table name
+     */
+    public void delete(final QualifiedName tableQName) {
+        final long start = registry.clock().wallTime();
+        try {
+            List<PartitionSequenceIds> partitionSequenceIds = getPartitionSequenceIds(tableQName);
+            while (!partitionSequenceIds.isEmpty()) {
+                _delete(partitionSequenceIds);
+                partitionSequenceIds = getPartitionSequenceIds(tableQName);
+            }
+        } finally {
+            this.fastServiceMetric.recordTimer(
+                HiveMetrics.TagDropHivePartitions.getMetricName(), registry.clock().wallTime() - start);
+        }
+    }
+
+    private List<PartitionSequenceIds> getPartitionSequenceIds(final QualifiedName tableQName) {
+        return jdbcTemplate.query(
+            String.format(SQL.PARTITIONS_SELECT_ALL, batchSize),
```

Comment:
is this the classmember batchSize? I would use this.batchSize to make it explicit.

Reply:
Will do.